# HG changeset patch
# User robm
# Date 1449750086 0
#      Thu Dec 10 12:21:26 2015 +0000
# Node ID b015efd2ab82fea24e06b539036c2e4978e26266
# Parent  c2ba29a6b98f628737ea1f74dd8745e4f72858a9
8145017: Add support for 3 digit hotspot minor version numbers
Reviewed-by: coffeys

diff --git a/src/share/native/common/jdk_util.c b/src/share/native/common/jdk_util.c
--- openjdk/jdk/src/share/native/common/jdk_util.c
+++ openjdk/jdk/src/share/native/common/jdk_util.c
@@ -53,7 +53,7 @@
     const char* jdk_update_string = JDK_UPDATE_VERSION;
     unsigned int jdk_update_version = 0;
     int len_update_ver = 0;
-    char update_ver[3];
+    char update_ver[5];
     char jdk_special_version = '\0';
 
     /* If the JDK_BUILD_NUMBER is of format bXX and XX is an integer
diff --git a/test/sun/misc/Version/Version.java b/test/sun/misc/Version/Version.java
--- openjdk/jdk/test/sun/misc/Version/Version.java
+++ openjdk/jdk/test/sun/misc/Version/Version.java
@@ -146,7 +146,7 @@
 
             String regex = "^([0-9]{1,2})";     // major
             regex += "\\.";                     // separator
-            regex += "([0-9]{1,2})";            // minor
+            regex += "([0-9]{1,3})";            // minor
             regex += "(\\-b([0-9]{1,3}))";      // JVM -bxx
             regex += ".*";
 
